wormhole-connect/src/hooks/useBalanceChecker.ts

```diff
@@ -0,0 +1,61 @@
+import { useEffect, useState, useMemo } from 'react';
+import { amount, routes } from '@wormhole-foundation/sdk';
+import { getWalletConnection, TransferWallet } from 'utils/wallet';
+import config from 'config';
+import { useIsMounted } from './useIsMounted';
+
+export const useBalanceChecker = (
+  quote?: routes.Quote<
+    routes.Options,
+    routes.ValidatedTransferParams<routes.Options>
+  >,
+): {
+  feeSymbol?: string;
+  isCheckingBalance: boolean;
+  isBalanceEnough: boolean;
+  walletBalance?: number;
+  networkCost?: number;
+} => {
+  const [isCheckingBalance, setCheckingBalance] = useState(false);
+  const [state, setState] = useState<{ balance: number; cost: number } | null>(
+    null,
+  );
+  const isMounted = useIsMounted();
+
+  const feeSymbol = useMemo(() => {
+    if (!quote?.relayFee?.token) return;
+
+    return config.sdkConverter.findTokenConfigV1(
+      quote?.relayFee?.token,
+      Object.values(config.tokens),
+    )?.symbol;
+  }, [quote]);
+
+  useEffect(() => {
+    setCheckingBalance(true);
+    (async () => {
+      try {
+        const wallet = getWalletConnection(TransferWallet.SENDING);
+        if (!wallet || !quote?.relayFee?.amount) return;
```

Comment:
You can check and return for `!quote?.relayFee?.amount` even before getting the wallet.